langchain4j-core/src/main/java/dev/langchain4j/model/chat/StreamingChatLanguageModel.java

```diff
@@ -1,26 +1,87 @@
 package dev.langchain4j.model.chat;
 
+import dev.langchain4j.Experimental;
 import dev.langchain4j.agent.tool.ToolSpecification;
 import dev.langchain4j.data.message.AiMessage;
 import dev.langchain4j.data.message.ChatMessage;
 import dev.langchain4j.data.message.UserMessage;
 import dev.langchain4j.model.StreamingResponseHandler;
+import dev.langchain4j.model.chat.request.ChatRequest;
+import dev.langchain4j.model.chat.request.ResponseFormat;
+import dev.langchain4j.model.chat.request.ResponseFormatType;
+import dev.langchain4j.model.chat.response.ChatResponse;
+import dev.langchain4j.model.chat.response.StreamingChatResponseHandler;
+import dev.langchain4j.model.output.Response;
 
 import java.util.List;
 
+import static dev.langchain4j.internal.Utils.isNullOrEmpty;
 import static java.util.Collections.singletonList;
 
 /**
+ * TODO review all javadoc in this class
  * Represents a language model that has a chat interface and can stream a response one token at a time.
+ *
+ * @see ChatLanguageModel
  */
 public interface StreamingChatLanguageModel {
 
+    /**
+     * TODO
+     * <p>
+     * A temporary default implementation of this method is necessary
+     * until all {@link StreamingChatLanguageModel} implementations adopt it. It should be removed once that occurs.
+     *
+     * @param chatRequest
+     * @param handler
+     */
+    @Experimental
+    default void chat(ChatRequest chatRequest, StreamingChatResponseHandler handler) {
+
+        ResponseFormat responseFormat = chatRequest.responseFormat();
+        if (responseFormat != null && responseFormat.type() == ResponseFormatType.JSON) {
```

Comment:
should it consider `supportedCapabilities()` instead?

Reply:
@langchain4j Honestly, in theory, I think it could be handy, but in practice, I'm not sure how much it would be really used by our users. I think where it's more handy is either indeed a stream of text tokens, and the other case is in the case of structured output, when we return a list of something (materialized by some POJO), streaming model could send each POJO as it's completed, so that a user interface, or some service consuming / subscribing to those POJOs could start treating those, without waiting for the full response. But then it's even more tricky to implement because models don't necessarily send stuff into the right coarse grained structures, so you'd have to accumulate until there's a POJO that can be read fully and sent back... It's too much hypothetical and added complexity, that I think it's not worse the effort, at least right now.

Reply:
What you describe actually works (for example in instructor library) and we have an issue for this: https://github.com/langchain4j/langchain4j/issues/910

But AFAIU your initial comment was about the generic support of structured outputs for streaming mode?